pages/ref/core-math.mdx

```diff
@@ -147,16 +147,28 @@ abs(-(-(-42))); // 42
 fun log(num: Int, base: Int): Int;
 ```
 
-Computes and returns the [logarithm](https://en.wikipedia.org/wiki/Logarithm) of a number `num` to the base `base`.
+Computes and returns the [logarithm](https://en.wikipedia.org/wiki/Logarithm) of a number `num` to the base `base`. Results are [rounded down](https://en.wikipedia.org/wiki/Rounding#Rounding_down). Passing negative `num` always gives $0$ as the result, while passing negative `base` gives $-1$ when it is close enough to zero and there's enough gas for calculations, otherwise throws an error with [exit code -14](/book/exit-codes): `Out of gas`.
 
-Usage example:
+Usage examples:
 
 ```tact
 log(1000, 10); // 3, as 10^3 is 1000
 //  ↑     ↑             ↑       ↑
 //  num   base          base    num
+
+log(1001, 10);  // 3
+log(999, 10);   // 2
+log(-1000, 10); // 0, because of the negative num
```

Comment:
this is a bug

Reply:
Logarithm of negative x's isn't defined, so this should throw errors, right? Or, should it return $-1$?

Reply:
yeah, it's either a complex number or it should throw

Reply:
Re-wrote the description as if this is merged already: https://github.com/tact-lang/tact/pull/342